src/Infrastructure/Service/Precondition/NoLinksPointOutsideTheCodebase.php

```diff
@@ -0,0 +1,52 @@
+<?php declare(strict_types=1);
+
+namespace PhpTuf\ComposerStager\Infrastructure\Service\Precondition;
+
+use PhpTuf\ComposerStager\Domain\Service\Precondition\NoLinksPointOutsideTheCodebaseInterface;
+use PhpTuf\ComposerStager\Domain\Value\Path\PathInterface;
+
+final class NoLinksPointOutsideTheCodebase extends AbstractLinkIteratingPrecondition implements
+    NoLinksPointOutsideTheCodebaseInterface
+{
+    public function getName(): string
+    {
+        return 'No links point outside the codebase'; // @codeCoverageIgnore
+    }
+
+    public function getDescription(): string
+    {
+        return 'The codebase cannot contain links that point outside the codebase.'; // @codeCoverageIgnore
+    }
+
+    protected function getFulfilledStatusMessage(): string
+    {
+        return 'There are no links that point outside the codebase.';
+    }
+
+    protected function getDefaultUnfulfilledStatusMessage(): string
+    {
+        return <<<'EOF'
+The %s directory at "%s" contains links that point outside the codebase, which is not supported. The first one is "%s".
+EOF;
+    }
+
+    protected function isSupportedLink(PathInterface $file, PathInterface $directory): bool
+    {
+        return !$this->linkPointsOutsidePath($file, $directory);
+    }
```

Comment:
Mmm, you're right: that is confusing. I was conflicted about introducing the term "codebase root directory", since it appears nowhere else. But upon reflection, I think it's probably the best term, so I added it to [the project glossary](https://github.com/php-tuf/composer-stager/wiki/Project-glossary) as a new domain concept and accepted your suggested change. Thanks!